CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
-* Support for "unorm8" is mandatory. All other encodings are optional.
-* When an unsupported encoding is requested, the encoding shall fall back to "unorm8".
-* The alpha channel is always interpreted as if clamped to [0,1].
-* Float RGB channel values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a (perceptually-linear) pixel value of (2,2,2) should be twice as bright as (1,1,1).
-* Operations on encoded values always operate on decoded values, not the encoded bits.
-    * I.e. in "unorm8-srgb" encoding, 0xff (1.0) minus 0xbc (0.5) equals 0xbc (0.5).
-
-
-#### Selecting the best color space match for the user agent's display device
-<pre>
-var colorSpace = window.matchMedia("(color-gamut: rec2020)").matches ? "rec2020" :
-    (window.matchMedia("(color-gamut: p3)").matches ? "p3" : "srgb");
-</pre>
-
-#### Selecting the best encoding for the user agent's display device
-Selection should be based on the best color space match (see above). For srgb, at least 8 bits per component is recommended; for p3, 10 bits; and for rec2020, 12 bits.  The float16 format is suitable for any colorspace.  There may soon be a proposal to add a way of detecting HDR displays, possibly something like "window.screen.isHDR()" (TBD), which would be a good hint to use the float16 format.
+Note that because there exists no direct access to the canvas backbuffer, it is not required that the colorType be truly respected. From the user's point of view, there will be no way to know if the true backing has higher precision than what was requested (e.g, the canvas is actually float32, when float16 was specified).
 
-#### Non-standard color spaces
-For future consideration: support could be added for color space defined using the [CSS @color-profile rule](https://www.w3.org/TR/css-color-4/#at-profile).
+### WebGL 2 API Changes
 
-#### Compositing the canvas element
-Canvas contents are composited in accordance with the canvas element's style (e.g. CSS compositing and blending rules). The necessary compositing operations must be performed in an intermediate colorspace, the compositing space, that is implementation specific. The compositing space must have sufficient precision and a sufficiently wide gamut to guarantee no undue loss of precision or gamut clipping in bringing the canvas's contents to the display. Implementations should not expose color spaces that are unreasonble for the display.
+IDL Additions:
+<pre>
+partial dictionary WebGLContextAttributes {
+  CanvasColorSpace colorSpace = "srgb";
+  CanvasColorType colorType = "uint8";
+  GLenum pixelFormat = GL_NONE;
```

Comment:
Propose this separately. What's the usecase for decoupling this from colorSpace/Type?

Reply:
2D canvas can't meaningfully support all formats (e.g, gl.SRGB8_ALPHA) in the same way that WebGL can. Also, 2D canvas has a lot of flexibility about what it does behind the scenes, because 2D canvas doesn't expose direct access to the framebuffer (e.g, one could request 10 bits per channel, and 2D canvas could lie and just use 16 bits per channel without much consequence, while for WebGL those details can leak through).